runtime/autoload/health.vim

```diff
@@ -29,7 +29,8 @@ function! health#check(plugin_names) abort
         \ ? s:discover_healthchecks()
         \ : s:get_healthcheck(a:plugin_names)
 
-  tabnew
+  " create scratch-buffer
+  execute 'tab sbuffer' nvim_create_buf(v:true, v:true)
```

Comment:
Hello. I believe this doesn't address the issue at hand. There's still no way to tell if this is a checkhealth buffer or not. Now it's a nofile buffer.

To solve the issue let's name the buffer to something meaningful. So we can close #16639 and  #8148
```suggestion
  let l:b = bufnr("nvim.chechhealth", v:true)
  execute 'tab sbuffer ' . l:b
  call nvim_buf_set_lines(l:b, 0, -1, v:true, [])
  setlocal buftype=nofile
  setlocal buflisted noswapfile
```

Reply:
@muniter, I believe setting both `&ft == markdown` and `&bt == nofile` should suffice for most practical purposes. About buffer name, it could bring more problems than it solves. For example, the user won't be able to run `:checkhealth nvim` and `:checkhealth provider` at the same time, because of buffer sharing issue. So if we want 100% reliable option then it should be a new filetype (and this is debatable if we really want it). While adding only `bt=nofile` is by all means very logical on its own.

Reply:
You can't do it at the same time with the current implementation, `:checkhealth` is a blocking call. That's why one would do `:checkhealth nvim provider`.

Reply:
By "the same time" I mean trying to open the second one while the first one is already open. As the name is fixed, second call to bufnr() will return an existing buffer id, so both commands will eventually share a single buffer.